# Практическая работа №3: Обработка выборочных данных. Нахождение интервальных оценок параметров распределения. Проверка статистической гипотезы о нормальном распределении

Выполнили студенты гр. 9383 Камзолов Никита и Крейсманн Кирилл. Вариант №3

## Цель работы

Получение практических навыков вычисления интервальных статистических оценок параметров распределения выборочных данных и проверки «справедливости» статистических гипотез.

## Основные теоретические положения

**Интервальной** называется статистическая оценка, которая задается двумя числами - концами интервала (в котором может находиться значение параметра $\Theta$)

**Точностью $\Theta^*$** - статистической оценки параметра $\Theta$ называется значение $\delta$, удоавлетворяющее неравенству: 
<p style="font-size: 14px; text-align: center;"> $\delta > |\Theta^* - \Theta|$ </p>

**Надежностью (доверительной вероятностью)** оценки $\Theta^*$ называется вероятность $\gamma$, с которой выполняется неравенство:
<p style="font-size: 14px; text-align: center;"> $\gamma = P\{\delta > |\Theta^* - \Theta|\}$ </p>

Интервал $(\Theta^* - \delta; \Theta^* + \delta)$ называется **доверительным интервалом** и является интервальной оценкой параметра $\Theta$

Значение $q = 1 - \gamma$ называют уровнем значимости.

**Интервальной оценкой математического ожидания при известном СКВО** служит доверительный интервал:
<p style="font-size: 14px; text-align: center;"> $\overline{x}_s - t\sigma/\sqrt N < a < \overline{x}_s + t\sigma/\sqrt N$, где </p>
$\overline{x}_s$ - выборочное среднеее - статистическая оценка математического ожидания.

$\sigma$ - среднекватратичное отклонение.

$N$ - размер выборки

$t = \delta\sqrt(N)/\sigma$

Уровень значимости можно посчитать с помощью функции Лапласа $\Phi$ (имеются табличные значения):
<p style="font-size: 14px; text-align: center;"> $P\{\overline{x}_s - t\sigma/\sqrt N < a < \overline{x}_s + t\sigma/\sqrt N\} = 2\Phi(t) = \gamma$ </p>

**Интервальной оценкой математического ожидания при неизвестном СКВО** служит доверительный интервал:

<p style="font-size: 14px; text-align: center;"> $\overline{x}_s - t_{\gamma}S/\sqrt N < a < \overline{x}_s + t_{\gamma}S/\sqrt N$, где </p>

$\overline{x}_s$ - выборочное среднеее - статистическая оценка математического ожидания.

$S$ - исправленное выборочное СКВО.

$N$ - размер выборки

$t_\gamma$ - может быть определено по заданным значениям $\gamma$ и $N$ по таблице.


**Интервальной оценкой среднего квадратического отклонения** служит неравенство:
<p style="font-size: 14px; text-align: center;"> $S(1-q) < \sigma < S(1+q)$, где </p>
$S$ - исправленное выборочное СКВО

$q$ - может быть определено по заданным значениям $\gamma$ и $N$ по таблице.

**Статистической гипотезой** называют гипотезу о виде неизвестного распределения исследуемой случайной величины или о значении параметров известных распределений.

Выдвинутую гипотезу называют **нулевой или основной гипотезой и обозначают $H_0$**

**Конкурирующей или альтернативной** называют гипотезу $H_1$, которая противоречит основной гипотезе.

**Ошибка первого рода** состоит в том, что нулевая гипотеза отвергается, хотя она верна.

**Ошибка второго рода** состоит в том, что нулевая гипотеза принимается, хотя она неверна.

Вероятность совершения ошибки первого рода называют **уровнем значимости** и обозначают через $\alpha$

Величину $1-\beta$ - вероятность того, что гипотеза отвергается, когда она неверна называется **мощностью критерия**

В качестве провекри гипотезы можно использовать критерий **$\chi^2$(Хи-квадрат) или иначе - критерий Пирсона.**

Согласно этому критерию на основании выборочных данных вычисляется "наблюденное" значение случайной величины $\chi^2$:

<p style="font-size: 14px; text-align: center;"> $\chi^2_{obs} = \sum_{i=1}^K \frac{(n_i - n_i^{'})^2}{n_i^{'}}$</p>

Здесь $n_i^{'}$ - теоретические частоты - частоты, с которыми нормально распределенная величина с параметрами $\overline{x}_s$ и $S$ попадала бы в $i$-ый интервал интервального ряда при проведении $N$ испытаний.

Распределение хи-квадрат зависит от числа **степеней свободы $k$**, которая в данном случае вычисляется по формуле $k = K - 3$, K - количество интервалов в интервальном ряде.

По числу степеней свободы и уровню значимости **вычисляется критическое значение хи-квадрат** $\chi^2_{crit}=\chi^2(\alpha, k)$. Значение можно определить из таблиц. Область принятия гипотезы $H_0$ определяется условием:
<p style="font-size: 14px; text-align: center;"> $\chi^2_{obs} < \chi^2_{crit}$ </p>

Для **вычисления "теоретических" частот** можно воспользоваться следующей последовательностью действий:
1. Пересчитываем значения границ интервалов интервального ряда. При этом левая граница первого интервала назначается равной $-\inf$, а правую границу последнего интервала назначаем равной $+\inf$. Остальные значения вычисляем по формуле:
<p style="font-size: 14px; text-align: center;"> $z_i = \frac{x_i - \overline{x}_s}{S}, i = 2, 3, ..., K$, где </p>
$x_i$ - граница интервала
2. Вычисляем вероятности попадания случайной величины в каждый $i$-ый инетрвал интервального ряда:
<p style="font-size: 14px; text-align: center;"> $p_i = \Phi(z_{i+1}) - \Phi(z_{i}), i = 1, 2, 3, ..., K$ </p>
3. Вычислить значения "теоретических" частот: 
<p style="font-size: 14px; text-align: center;"> $n_i^{'}=N*p_i, i = 1, 2, 3, ..., K$ </p>


## Постановка задачи

Для заданной надежности определить (на основании выборочных данных и результатов выполнения практической работы №2) границы доверительных интервалов для математического ожидания и среднеквадратичного отклонения случайной величины. Проверить гипотезу о нормальном распределении исследуемой случайной величины с помощью критерия Пирсона $\chi^2$. Дать содержательную интерпретацию полученным результатам.

## Выполнение работы

In [4]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import math

При выполнении практических работ 1 и 2 были получены результаты, которые можно использовать в данной работе.

Интервальный ряд:

|   | $[x_{i}, x_{i+1})$ | $\tilde{x}_{i}$ | $n_i$ | $p^*_i$ | $n_{i}^{\Sigma}$ | $p_{i}^{\Sigma}$ |
|:-:|:------------------:|:-------------:|:-----:|:-----:|:--------------:|:--------------:|
| 1 |    $[344, 379)$    |      361.5      |    11   | 0.102804  |        11       |       0.102804     |
| 2 |    $[379, 414)$    |      396.5     |  13  | 0.121495 |       24       |      0.224299     |
| 3 |    $[414, 449)$    |      431.5      |   35| 0.327103 |       59       |      0.551402     |
| 4 |    $[449, 484)$    |      466.5     |    20 | 0.186916 |        79      |      0.738318     |
| 5 |    $[484, 519)$    |      501.5     |   15  | 0.140187 |       94       |      0.878505     |
| 6 |    $[519, 554)$    |      536.5     |   9  | 0.0841121 |       103      |      0.962617     |
| 7 |    $[554, 585]$    |      569.5     |   4   | 0.0373832 |       107      |      1     |
|||$\sum$|107|1.0|-|-|

Количество интервалов k = 7; шаг h = 35; размер выборки N = 107.

In [5]:
k=7
h=35
N=107

In [6]:
left_bounds = [344,379,414,449,484,519,554]#левые границы интервалов
middle_points = [361.5, 396.5, 431.5, 466.5, 501.5, 536.5, 569.5] #середина i-го частичного интервала
absolute_freq = [11,13,35,20,15,9,4]#абсолютная частота i-го частичного интервала
relative_freq = [0.102804,0.121495,0.327103,0.186916,0.140187,0.0841121,0.0373832]#относительная частота i-го интервала
accum_relative_freq = [0.102804,0.224299,0.551402,0.738318,0.878505,0.962617,1]#накопленная частота для i-го интервала
intervals = ['[344, 379]', '(379, 414]', '(414, 449]', '(449, 484]', '(484, 519]', '(519, 554]', '(554, 585]']

Выборочное среднее $\overline{x}_s = \frac{1}{N} \sum n_j x_j = 450.397$

Выборочная дисперсия $D_s = \frac{1}{N} \sum{n_j (x_j - \overline{x_s})^2} = 2850.615$

Среднеквадратичное отклонение $\sigma = \sqrt{D_s} = 53.391$

Исправленная выборочная дисперсия $S^2 = \frac{N}{N-1} D_s = 2877.508$

Исправленное среднеквадратичное отклонение $S = \sqrt{s^2} = 53.642$

In [7]:
standart_mean = 450.397 #выборочное среднее
Ds = 2850.615 #выборочная дисперсия
sigma = 53.391 #среднеквадратичное отклонение
S2 = 2877.508 #исправленная выборочная дисперсия
S = 53.642 #исправленное среднеквадратичное отклонение

### 1. Вычислить точность и доверительный интервал для математического ожидания при неизвестном среднеквадратичном отклонении при заданном объёме выборки для доверительной точности $\gamma \in \{0.95, 0.99\}$. Сделать выводы.

Проведем вычисления для доверительной точности $\gamma \in \{0.95, 0.99\}$

Вычисления проиводим по формуле: $\overline{x}_s - t_{\gamma}S/\sqrt N < a < \overline{x}_s + t_{\gamma}S/\sqrt N$

$t_{\gamma_{0.95}} = 1.983$ (из таблицы, при $\gamma = 0.95, N = 107$)

$t_{\gamma_{0.99}} = 2.623$ (из таблицы, при $\gamma = 0.99, N = 107$)

In [12]:
t_gamma_0_95 = 1.983
left = standart_mean - t_gamma_0_95 * S / math.sqrt(N)
right = standart_mean + t_gamma_0_95 * S / math.sqrt(N)
print("Доверительный интервал при доверительной точности 0.95: ({}, {})".format(left, right))

t_gamma_0_99 = 2.623
left = standart_mean - t_gamma_0_99 * S / math.sqrt(N)
right = standart_mean + t_gamma_0_99 * S / math.sqrt(N)
print("Доверительный интервал при доверительной точности 0.99: ({}, {})".format(left, right))

Доверительный интервал при доверительной точности 0.95: (440.11362230478966, 460.6803776952103)
Доверительный интервал при доверительной точности 0.99: (436.79473086508483, 463.99926913491515)


Из полученного результата можем сделать вывод, что статистическая оценка математического ожидания попадает в интервал (440.1136, 460.6804) с вероятностью 0.95, и в интервал (436.7947, 463.9992) с вероятностью 0.99.

### 2. Для вычисления границ доверительного интервала для среднеквадратичного отклонения определить значение $q$ при заданных $\gamma$ и $N$. Построить доверительные интервалы, сделать выводы.

Проведем вычисления для доверительной точности $\gamma \in  \{0.95, 0.99\}$

Вычисления проиводим по формуле: $S(1-q)<\sigma<S(1+q)$

$q_{0.95} = 0.139$ (из таблицы, при $\gamma = 0.95, N = 107$)

$q_{0.99} = 0.193$ (из таблицы, при $\gamma = 0.99, N = 107$)

In [14]:
q_0_95 = 0.139
left = S * (1 - q_0_95)
right = S * (1 + q_0_95)
print("Доверительный интервал при доверительной точности 0.95: ({}, {})".format(left, right))

q_0_99 = 0.193
left = S * (1 - q_0_99)
right = S * (1 + q_0_99)
print("Доверительный интервал при доверительной точности 0.99: ({}, {})".format(left, right))

Доверительный интервал при доверительной точности 0.95: (46.185762000000004, 61.098238)
Доверительный интервал при доверительной точности 0.99: (43.289094, 63.99490600000001)


Из полученного результата можем сделать вывод, что статистическая оценка cредневадратичного отклонения попадает в интервал (46.1857, 61.0982) с вероятностью 0.95, и в интервал (43.2890, 63.9949) с вероятностью 0.99.

### 3. Проверить гипотезу о нормальности заданного распределения с помощью критерия $\chi^2$ (Пирсона). Для этого необходимо найти теоретические частоты и вычислить наблюдаемое значение критерия. Для удобства вычисления необходимо заполнить таблицу. Проконтролировать корректность вычисления $\chi^2_{obs}$. Далее по заданному уровню значимости $\alpha = 0.05$ и числу степеней свободы $df$ найти критическую точку $\chi^2_{crit}$ и сравнить с наблюдаемым значением. Сделать выводы.


In [15]:
z_i = ["-inf"] #пересчитываем значения границ интервалов интервального ряда 
for i in range(1, len(left_bounds)):
    z_i.append((left_bounds[i] - standart_mean) / S)
z_i.append("+inf")
print(z_i)

['-inf', -1.3309906416613846, -0.678516833824242, -0.026043025987099497, 0.626430781850043, 1.2789045896871856, 1.931378397524328, '+inf']


С помощью таблицы 4 и знания о том, что $\Phi(\infty) = 0.5$ запишем значения функции $\Phi(z_i), i = 1, ..., K + 1$

$\Phi_i = [-0.5, -0.4082, -0.2486, -0.0080, 0.2324, 0.3979, 0.4732, 0.5]$

In [16]:
Phi_i = [-0.5, -0.4082, -0.2486, -0.0080, 0.2324, 0.3979, 0.4732, 0.5]

In [17]:
p_i = [] #вероятности попадания случайной величины в каждый 𝑖 -ый интервал 
n_i_stroke = [] #значения "теоретических" частот
ni_nistroke_2 = [] #значение выражения (n_i - n_i^')^2
ni_nistroke_2_nistroke = [] #значение выражения (n_i - n_i^')^2/n_i^'
n_i_2 = [n ** 2 for n in absolute_freq] #абсолютная частота в квадрате
ni2_nistroke = [] #значение выражения n_i^2/n_i^'

for i in range(len(Phi_i) - 1):
    p_i.append(Phi_i[i+1] - Phi_i[i])
    n_i_stroke.append(N * p_i[i])
    ni_nistroke_2.append((absolute_freq[i] - n_i_stroke[i]) ** 2)
    ni_nistroke_2_nistroke.append(ni_nistroke_2[i]/n_i_stroke[i])
    ni2_nistroke.append(n_i_2[i] / n_i_stroke[i])

In [19]:
#подсчет сумм
#print("$\\sum n_i$ =", sum(absolute_freq), "\n")
#print("$\\sum p_i$ =", sum(p_i), "\n")
#print("$\\sum n_i^{'}$ =", sum(n_i_stroke), "\n")
#print("$\\frac{(n_i - n_i^{'})^2}{n_i^{'}}$ =", sum(ni_nistroke_2_nistroke), "\n")
#print("$\\frac{n_i^2}{n_i^{'}}$ =", sum(ni2_nistroke), "\n")

$\sum n_i$ = 107 

$\sum p_i$ = 1.0 

$\sum n_i^{'}$ = 107.0 

$\frac{(n_i - n_i^{'})^2}{n_i^{'}}$ = 6.687293148829535 

$\frac{n_i^2}{n_i^{'}}$ = 113.68729314882953 

In [ ]:
table = pd.DataFrame(
    {
        "$[x_i, x_{i+1})$": intervals,#интервалы
        "$n_i$": absolute_freq,#абсолютные частоты
        "$p_i$": p_i,#вероятности попадания случайной величины в каждый 𝑖 -ый интервал 
        "$n_i^{'}$": n_i_stroke,#значения "теоретических" частот
        "$(n_i - n_i^{'})^2$": ni_nistroke_2,#значение выражения (n_i - n_i^')^2
        "$\\frac{(n_i - n_i^{'})^2}{n_i^{'}}$": ni_nistroke_2_nistroke,#значение выражения (n_i - n_i^')^2/n_i^'
        "$n_i^2$": n_i_2,#абсолютная частота в квадрате
        "$\\frac{n_i^2}{n_i^{'}}$": ni2_nistroke#значение выражения n_i^2/n_i^'
    }
)
table.index = table.index + 1
#print(table.to_markdown())   

|    | $[x_i, x_{i+1})$   |   $n_i$ |   $p_i$ |   $n_i^{'}$ |   $(n_i - n_i^{'})^2$ |   $\frac{(n_i - n_i^{'})^2}{n_i^{'}}$ |   $n_i^2$ |   $n_i^2/n_i^{'}$ |
|---:|:-------------------|--------:|--------:|------------:|----------------------:|--------------------------------------:|----------:|------------------:|
|  1 | [344, 379]         |      11 |  0.0918 |      9.8226 |               1.38627 |                              0.141131 |       121 |          12.3185  |
|  2 | (379, 414]         |      13 |  0.1596 |     17.0772 |              16.6236  |                              0.973436 |       169 |           9.89624 |
|  3 | (414, 449]         |      35 |  0.2406 |     25.7442 |              85.6698  |                              3.32773  |      1225 |          47.5835  |
|  4 | (449, 484]         |      20 |  0.2404 |     25.7228 |              32.7504  |                              1.27321  |       400 |          15.5504  |
|  5 | (484, 519]         |      15 |  0.1655 |     17.7085 |               7.33597 |                              0.414263 |       225 |          12.7058  |
|  6 | (519, 554]         |       9 |  0.0753 |      8.0571 |               0.88906 |                              0.110345 |        81 |          10.0532  |
|  7 | (554, 585]         |       4 |  0.0268 |      2.8676 |               1.28233 |                              0.447179 |        16 |           5.57958 |
|$\sum$|          -|     107 |       1 |         107 |   -                   |                              6.687293 | -         |          113.6872 |

<p style="font-size: 14px; text-align: center;"> $\chi^2_{obs} = \sum_{i=1}^K \frac{(n_i - n_i^{'})^2}{n_i^{'}} = 6.687$</p>

Контроль вычислений:
<p style="font-size: 14px; text-align: center;"> $\chi^2_{obs} = \sum_{i=1}^K \frac{n_i^2}{n_i^{'}} - 2 \cdot \sum_{i=1}^K n_i + \sum_{i=1}^K n_i^{'} = \sum_{i=1}^K \frac{n_i^2}{n_i^{'}} - n = 113.687 - 107 = 6.687$</p>

<p style="font-size: 14px; text-align: center;">Число степеней работы $k = K - 3 = 4$ </p>
<p style="font-size: 14px; text-align: center;"> $\chi^2_{krit} = 9.5$ </p>
<p style="font-size: 14px; text-align: center;"> $\chi^2_{obs} < \chi^2_{crit}$ </p>

На основе полученных результатов можем сделать вывод, что данные принимаются гипотезой $\chi^2$ и имеют распределение, близкое к нормальному.

## Выводы

В ходе выполнения практической работы были получены границы доверительных интервалов для оценки математического ожидания при неизвестном СКВО. Статистическая оценка математического ожидания попадает в интервал (440.1136, 460.6804) с вероятностью 0.95, и в интервал (436.7947, 463.9992) с вероятностью 0.99.

Также были получены границы доверительных интервалов для оценки среднего квадратического отклонения. Статистическая оценка cредневадратичного отклонения попадает в интервал (46.1857, 61.0982) с вероятностью 0.95, и в интервал (43.2890, 63.9949) с вероятностью 0.99.

Была проверена гипотеза о нормальности заданного распределения с помощью критерия $\chi^2$ с заданным уровнем значимости $\alpha = 0.05$. Получив неравенство $\chi^2_{obs} < \chi^2_{crit}$, делаем вывод о принятии гипотезы.